packages/fuel-indexer-database/postgres/src/lib.rs

```diff
@@ -757,40 +757,113 @@ pub async fn revert_transaction(
+    execute_query(
+        conn,
+        format!(
+            "DELETE FROM graph_registry_columns WHERE type_id IN (SELECT id FROM graph_registry_type_ids WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}');"
+        ),
+    )
+    .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_asset_registry_wasm WHERE index_id = {index_id}"),
+        format!(
+            "DELETE FROM graph_registry_type_ids WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}';"
+        ),
     )
     .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_asset_registry_manifest WHERE index_id = {index_id}"),
+        format!(
+            "DELETE FROM graph_registry_root_columns WHERE root_id = (SELECT id FROM graph_registry_graph_root WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}');"
+        ),
     )
     .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_asset_registry_schema WHERE index_id = {index_id}"),
+        format!(
+            "DELETE FROM graph_registry_graph_root WHERE schema_name = '{namespace}' AND schema_identifier = '{identifier}';"
+        ),
     )
     .await?;
 
     execute_query(
         conn,
-        format!("DELETE FROM index_registry WHERE id = {index_id}"),
+        format!("DROP SCHEMA {namespace}_{identifier} CASCADE"),
     )
     .await?;
 
     Ok(())
 }
 
+#[cfg_attr(feature = "metrics", metrics)]
+pub async fn remove_indexers(
+    conn: &mut PoolConnection<Postgres>,
+    namespace: &str,
+    identifier: &str,
+    limit: Option<usize>,
+) -> sqlx::Result<()> {
+    let limit = limit.map(|l| format!("LIMIT {l}")).unwrap_or_default();
+
+    execute_query(
+        conn,
+        format!(
+            "DELETE FROM index_asset_registry_wasm WHERE index_id IN
+            (SELECT id FROM index_registry
+                WHERE namespace = '{namespace}' AND identifier = '{identifier}' {limit})"
+        ),
+    )
+    .await?;
+
+    execute_query(
+        conn,
+        format!(
+            "DELETE FROM index_asset_registry_manifest WHERE index_id IN
+            (SELECT id FROM index_registry
+                WHERE namespace = '{namespace}' AND identifier = '{identifier}' {limit})"
+        ),
+    )
+    .await?;
+
+    execute_query(
+        conn,
+        format!(
+            "DELETE FROM index_asset_registry_schema WHERE index_id IN
+            (SELECT id FROM index_registry
+                WHERE namespace = '{namespace}' AND identifier = '{identifier}' {limit})"
+        ),
+    )
+    .await?;
+
+    execute_query(
+        conn,
+        format!(
+            "DELETE FROM index_registry WHERE id IN
+            (SELECT id FROM index_registry
+                WHERE namespace = '{namespace}' AND identifier = '{identifier}' {limit})"
+        ),
+    )
+    .await?;
+
+    Ok(())
+}
+
+#[cfg_attr(feature = "metrics", metrics)]
+pub async fn remove_indexer(
```

Comment:
- Any reason that `remove_indexers` is broken out? 
- `remove_indexers` implies that we'll remove multiple indexers, but as the code, reads, we're only ever removing a single one ?